x/gamm/keeper/swap.go

```diff
@@ -83,12 +84,13 @@ func (k Keeper) SwapExactAmountOut(
 	tokenInDenom string,
 	tokenInMaxAmount sdk.Int,
 	tokenOut sdk.Coin,
+	swapFeeMultiplier sdk.Dec,
 ) (tokenInAmount sdk.Int, err error) {
 	pool, err := k.getPoolForSwap(ctx, poolId)
 	if err != nil {
 		return sdk.Int{}, err
 	}
-	swapFee := pool.GetSwapFee(ctx)
+	swapFee := pool.GetSwapFee(ctx).Mul(swapFeeMultiplier)
```

Comment:
I think we should protect against negative numbers here (ensuring it is not possible to set a negative swap fee since we are now essentially modifying the pools swap fee). What do you think?

Reply:
I think this place is not where we should force and check for negative swaps fee. If we have here negative swap fee, than something is broken prior this code.

Or you mean, that `swapFeeMultiplier` can't be negative. Is so, that make sense, will add one more check here.

Reply:
Check out new version, where i remove this input param. So it is not actual anymore.